src/market/gtc.ts

```diff
@@ -138,18 +200,79 @@ async function estimateGasBuy(
  * @param price - The price of the order.
  * @param size - The size of the order.
  * @param postOnly - A boolean indicating whether the order should be post-only.
+ * @param txOptions - The transaction options for the order.
  * @returns A promise that resolves to a boolean indicating success or failure.
  */
 async function addSellOrder(
     orderbook: ethers.Contract,
     price: BigNumber,
     size: BigNumber,
-    postOnly: boolean
+    postOnly: boolean,
+    txOptions?: TransactionOptions
 ): Promise<ContractReceipt> {
+    console.time('Total Limit Sell Time');
     try {
-        const tx = await orderbook.addSellOrder(price, size, postOnly);
-        return await tx.wait();
+        console.time('Get Signer Time');
+        const signer = orderbook.signer;
+        const address = await signer.getAddress();
+        console.timeEnd('Get Signer Time');
+
+        const data = orderbook.interface.encodeFunctionData("addSellOrder", [
+            price,
+            size,
+            postOnly
+        ]);
+
+        const tx: ethers.providers.TransactionRequest = {
+            to: orderbook.address,
+            from: address,
+            data,
+            ...(txOptions?.nonce !== undefined && { nonce: txOptions.nonce }),
+            ...(txOptions?.gasLimit && { gasLimit: txOptions.gasLimit }),
+            ...(txOptions?.gasPrice && { gasPrice: txOptions.gasPrice }),
+            ...(txOptions?.maxFeePerGas && { maxFeePerGas: txOptions.maxFeePerGas }),
+            ...(txOptions?.maxPriorityFeePerGas && { maxPriorityFeePerGas: txOptions.maxPriorityFeePerGas })
+        };
+
+        console.time('RPC Calls Time');
```

Comment:
should the benchmarking console logs exist?

Reply:
I am pushing without it rn